testcommon/mockSmartContractCallerTest.go

```diff
@@ -0,0 +1,131 @@
+package testcommon
+
+import (
+	"testing"
+
+	"github.com/ElrondNetwork/arwen-wasm-vm/arwen"
+	mock "github.com/ElrondNetwork/arwen-wasm-vm/mock/context"
+	worldmock "github.com/ElrondNetwork/arwen-wasm-vm/mock/world"
+	"github.com/ElrondNetwork/elrond-go/core/vmcommon"
+)
+
+type testSmartContract struct {
```

Comment:
Optionally make a separate file `mockTestSmartContract.go` containing this structure along with `MockTestSmartContract` and its methods.

Reply:
done